xarray/tests/test_backends.py

```diff
@@ -106,23 +106,58 @@
 except ImportError:
     pass
 
-have_zarr_kvstore = False
+
 try:
-    from zarr.storage import KVStore
+    import zarr
+    import zarr.codecs
+except ImportError:
+    have_zarr = False
+else:
+    have_zarr = True
 
+
+if have_zarr:
```

Comment:
Our usual pattern is to define `has_zarr`, `requires_zarr`, `has_zarr_v3` and `requires_zarr_v3` in `tests/__init__.py` (there's a helper function there)